From 993b15ca042f746abaa211c6ec726860647728bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:26:39 +0800
Subject: [PATCH] Revert "vmalloc: remove BUG_ON due to racy counting of
 VM_LAZY_FREE"

This reverts commit 21ba0cac4b63286aaf41befb3b93c8ffb5036306.
---
 mm/vmalloc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index c85d75406c81..0f551a4a44cd 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -555,8 +555,10 @@ static void __purge_vmap_area_lazy(unsigned long *start, unsigned long *end,
 	}
 	rcu_read_unlock();
 
-	if (nr)
+	if (nr) {
+		BUG_ON(nr > atomic_read(&vmap_lazy_nr));
 		atomic_sub(nr, &vmap_lazy_nr);
+	}
 
 	if (nr || force_flush)
 		flush_tlb_kernel_range(*start, *end);
-- 
2.35.3

